File: python/0058_length_of_last_word.py
# Problem 58: Length of Last Word
# https://leetcode.com/problems/length-of-last-word/

class Solution:
    def length_of_last_word(self, s: str) -> int:
        # 利用 split 把字串拆成陣列
        word_list = s.split()
        # index -1 取 最後一個值, 再用 len 取得長度
        return len(word_list[-1])

File: go/0058_length_of_last_word.go
// Problem 58: Length of Last Word
// https://leetcode.com/problems/length-of-last-word/

package Go

import "strings"

//lint:file-ignore U1000 Ignore all unused code
func lengthOfLastWord(s string) int {
	// 去除頭尾的空白
	s = strings.TrimSpace(s)
	// 使用空白字元拆分字串成單字陣列
	words := strings.Fields(s)
	// 回傳最後一個單字的長度
	return len(words[len(words)-1])
}
